src/Microsoft.Health.Fhir.TemplateManagement/Overlay/OverlayFileSystem.cs

```diff
@@ -7,99 +7,115 @@
 using System.IO;
 using System.Linq;
 using EnsureThat;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
 using Microsoft.Health.Fhir.TemplateManagement.Models;
 
 namespace Microsoft.Health.Fhir.TemplateManagement.Overlay
 {
     public class OverlayFileSystem : IOverlayFileSystem
     {
+        private readonly string _workingFolder;
```

Comment:
add some comments and samples

Reply:
Add comments for functions in interface IOverlayFileSystem.